doc/memos/rdm-draft-alamos-lanzieri-runtime-configuration-architecture.md

```diff
@@ -0,0 +1,444 @@
+The configuration manager is a pseudo-module that allows a RIOT node to be
+configured from one or more communication interfaces. Examples of these
+communication interfaces could be UART, SPI or higher layers like PPP, IPv6,
+UDP, CoAP, etc. 
+
+These are some examples of configuration mechanisms that could be provided by 
+the Configuration Manager
+- A UART shell with special commands for interacting with configurations
+- CoAP configuration resources
+- An NFC entrypoint for configuring RIOT from a master NFC device.
+
+The Configuration Manager may interact with the RIOT
+Registry for interacting with persistent configurations.
+
+The Configuration Manager MAY be replaced by any existing Device Management
+system like [Newt Manager](https://mynewt.apache.org/master/os/modules/devmgmt/newtmgr.html)
+ or [OMA LWM2M](https://www.omaspecworks.org/what-is-oma-specworks/iot/lightweight-m2m-lwm2m/)
+
+The configuration manager SHOULD provide an access control mechanism for
+restricting access to configurations or restricting configuration interfaces.
+Extra security can be implemented in lower layers like the ones mentioned above
+(e.g CoAP and DTLS, CHAP on PPP, etc)
+
+The Configuration Manager API MUST provide helpers to modify access control
+settings as well as enabling/disabling communication interfaces.
+
+## Acknowledgements
+
+## References
+- [Configuration Task
+Force](https://github.com/RIOT-OS/RIOT/wiki/Configuration-Task-Force-(CTF))
+- [Mynewt OS config module
+  documentation](https://mynewt.apache.org/latest/os/modules/config/config.html)
+- [Newt Manager](https://mynewt.apache.org/master/os/modules/devmgmt/newtmgr.html)
+- [OMA LWM2M](https://www.omaspecworks.org/what-is-oma-specworks/iot/lightweight-m2m-lwm2m/)
+
+## Revisions
+- Rev0: initial document
+
+## Contact
+The authors of this memo can be contacted via email at
+jose.alamos@haw-hamburg.de and leandro.lanzieri@haw-hamburg.de
+
+## Appendix
+### Conceptual example of Registry Handler
+
+```c
+/*
+This is a conceptual example of a "my_handler" Registry Handler that exposes `threshold` and `is_enabled` parameters.
+For education purposes, it will be assumed the application crashes if a `set_threshold` function is called
+when `is_enabled==false`.
+
+Note the Registry Handler is not aware of any storage mechanism.
+*/
+
+/* Define a registry handler for the current RIOT module.
+   To be registered in the RIOT Registry */
+registry_handler_t my_handler = {
+    .name = "my_handler",
+    .get = my_get_handler,
+    .set = my_set_handler,
+    .commit = my_commit_handler,
+    .export = my_export_handler
+};
+
+/* These are the state variables and parameters defined for this module */
+static int is_enabled = false;
+static int threshold = 0;
+
+/* ... */
+
+/* Dummy implementation of `get` handler.
+   For both configuration parameters, it copies the value to a `val` variable.
+*/
+char *my_get_handler(int argc, char **argv, char *val, int val_len_max)
```

Comment:
The original idea was to return either a pointer to `val` or NULL in case of error. But since `val` is already there, maybe we can change the prototype to return error messages